module/modmanager/manager.go

```diff
@@ -1172,6 +1177,21 @@ func (m *module) registerResources(mgr modmaninterface.ModuleManager, logger log
 					) (resource.Resource, error) {
 						return mgr.AddResource(ctx, conf, DepsToNames(deps))
 					},
+					Discover: func(ctx context.Context, logger logging.Logger) (interface{}, error) {
+						req := &robotpb.DiscoverComponentsRequest{
+							Queries: []*robotpb.DiscoveryQuery{
+								{Subtype: api.API.String(), Model: model.String()},
+							},
+						}
+
+						res, err := m.robotClient.DiscoverComponents(ctx, req)
+						if err != nil {
+							m.logger.Errorf("error in modular DiscoverComponents: %w", err)
+							return nil, err
+						}
+
+						return res, nil
+					},
```

Comment:
non-urgent, we're only adding this for modular components, do we want to add it for modular services as well? There is no difference for in-rdk resources.

Reply:
Going to do this in the future https://viaminc.slack.com/archives/C06542APFRQ/p1727891009519449?thread_ts=1727817423.037039&cid=C06542APFRQ